src/compiler/transformers/component-lazy/form-internals.ts

```diff
@@ -0,0 +1,47 @@
+import type * as d from '@stencil/core/declarations';
+import ts from 'typescript';
+
+import { HOST_REF_ARG } from './constants';
+
+/**
+ * Create a binding for an `ElementInternals` object compatible with a
+ * lazy-load ready Stencil component.
+ *
+ * In order to create a lazy-loaded form-associated component we need to access
+ * the underlying host element (via the "$hostElement$" prop on {@link d.hostRef})
+ * to make the `attachInternals` call on the right element. This means that the
+ * code generated by this function depends on their being a variable in scope
+ * called {@link HOST_REF_ARG} with type {@link HTMLElement}.
+ *
+ * @param cmp metadata about the component of interest, gathered during compilation
+ * @returns a list of expression statements
+ */
+export function createLazyFormInternalsBinding(cmp: d.ComponentCompilerMeta): ts.ExpressionStatement[] {
+  if (cmp.formAssociated && cmp.formInternalsMemberName) {
+    return [
+      ts.factory.createExpressionStatement(
```

Comment:
Any value in a comment of what this expression looks like in generated TS? Factory AST stuff isn't the most friendly to read 😂

Reply:
makes sense! especially the one here for the lazy build isn't so intuitive. Adding a comment here (and so the native component one) which explain what we're trying to generate